app/beacon-omop-worker/scripts/worker.py

```diff
@@ -67,14 +86,27 @@ def main() -> None:
         drivername=os.getenv("DATASOURCE_DB_DRIVERNAME", config.DEFAULT_DB_DRIVER),
         schema=os.getenv("DATASOURCE_DB_SCHEMA"),
     )
-    #
+    output_file_name = "output.json"
     if args.command == "filteringterms":
-        output_file_name = "output.json"
         logger.info("Extracting filtering terms...")
         filtering_terms = query_solvers.solve_filters(db_manager=db_manager)
         try:
             logger.info("Saving filtering terms to file...")
-            save_to_output(filtering_terms, output_file_name)
+            save_filtering_terms(filtering_terms, output_file_name)
             logger.info(f"Saved results to {output_file_name}")
         except ValueError as e:
             logger.error(str(e), exc_info=True)
+
```

Comment:
At the end of the `if` blocks you can add and `exit()` statement to end the program earlier and not run anything that might not be related to the command the user issued.